### PR TITLE
Ticket - Transaction List

### PR DESCRIPTION
**Pull request by ChristianMoschke**
[Ticket - Transaction List](https://github.com/miToe/dagobert/issues/7)

### CODE REVIEW

File: src/components/TransactionList/TransactionList.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import React from 'react';
```
Comment (doemser) [inline src/components/TransactionList/TransactionList.js:1]: you do not need to import React

File: src/components/TransactionList/TransactionList.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+// Example list of transactions
+// TODO: to be removed and saved into localStorage
```
Comment (doemser) [inline src/components/TransactionList/TransactionList.js:4]: when introducing localStorage you will still need this data in you project, you will just replace a useState with a useLocalStorageState and then need initial data as well.

File: src/components/TransactionList/TransactionList.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+const transactions = [
```
Comment (doemser) [inline src/components/TransactionList/TransactionList.js:5]: I would recommend placing it somewheere like:

`@/lib/transactions.js`

File: src/components/TransactionList/TransactionList.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  {
+    "Date": "2023-12-01",
+    "Amount": 100.50,
+    "Currency": "USD",
+    "Category": "Groceries",
+    "Description": "Weekly grocery shopping",
+    "PaymentMethod": "Credit Card"
```
Comment (doemser) [inline src/components/TransactionList/TransactionList.js:12]: use lowercase keys in objects like so:

```javascript
{
date: "2023-12-01",
amount: 100.50,
}
```

File: src/components/TransactionList/TransactionList.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    "Date": "2023-11-27",
+    "Amount": 120.00,
+    "Currency": "USD",
+    "Category": "Dining",
+    "Description": "Brunch with friends",
+    "PaymentMethod": "Credit Card"
+  }
+];
+
+const TransactionList = () => {
```
Comment (doemser) [inline src/components/TransactionList/TransactionList.js:88]: If you use ChatGPT or similar to support you, make sure to review the code to align with our coding practices.

```javascript
function TransActionList() {}
```

File: src/components/TransactionList/TransactionList.js (hunk 4 above)
Comment (doemser) [reply src/components/TransactionList/TransactionList.js:12]: also add an id to your dataset

File: src/components/TransactionList/TransactionList.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  // Sort transactions by date in descending order
+  const sortedTransactions = transactions.sort((a, b) => new Date(b.Date) - new Date(a.Date));
+
+  return (
+    <div>
+      <h1>Transactions</h1>
+      <ul>
+        {sortedTransactions.map((transaction, index) => (
+          <li key={index}>
```
Comment (doemser) [inline src/components/TransactionList/TransactionList.js:97]: Never use an index as key in react, instead use the id of the dataset

```suggestion
        {sortedTransactions.map((transaction) => (
          <li key={transaction.id}>
```

File: src/components/TransactionList/TransactionList.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+            <h3>{transaction.Category}</h3>
+            <div>{new Date(transaction.Date).toLocaleDateString('de-DE')}</div>
+            <div>{transaction.Amount.toFixed(2)} {transaction.Currency}</div>
```
Comment (doemser) [inline src/components/TransactionList/TransactionList.js:100]: Keep correctly rounded amounts in you dataset instead

```suggestion
            <div>{transaction.amount} {transaction.currency}</div>
```

File: src/pages/index.js [hunk 8]
```diff
@@ -1,7 +1,12 @@
-export default function HomePage() {
+import React from 'react';
```
Comment (doemser) [inline src/pages/index.js:1]: you do not need this imports

File: src/pages/index.js [hunk 9]
```diff
@@ -1,7 +1,12 @@
...
+import TransactionList from "@/src/components/TransactionList/TransactionList";
+
+const Home = () => {
```
Comment (doemser) [inline src/pages/index.js:4]: replace this whole page with the TransActionList Component.

**Review by doemser — CHANGES_REQUESTED** [review]
Good first ticket, just found a few things ;)

File: src/pages/index.js [hunk 10]
```diff
@@ -1,7 +1,23 @@
...
   return (
     <div>
-      <h1>Hello from Next.js</h1>
+      <h1>Transactions</h1>
+      <ul>
+        {sortedTransactions.map((transaction) => (
+          <li key={transaction.id}>
+            <h3>{transaction.category}</h3>
+            <div>{new Date(transaction.date).toLocaleDateString('de-DE')}</div>
+            <div>{transaction.amount.toFixed(2)} {transaction.currency}</div>
```
Comment (Braun-Johannes) [inline src/pages/index.js:15]: Old suggestion:

Keep correctly rounded amounts in you dataset instead
 <div>{transaction.amount} {transaction.currency}</div>

**Review by Braun-Johannes — CHANGES_REQUESTED** [review]
So far the changes are good. You missed one suggestion from the old review. I marked this one again

**Review by Braun-Johannes — APPROVED** [review]
LGTM

**Pull request merged**